js/__internal/grids/grid_core/columns_controller/module.ts

```diff
@@ -1,2556 +1,1769 @@
 
-          return result;
-        },
-        getInvisibleColumns(columns, bandColumnIndex) {
-          const that = this;
-          let result: any = [];
-          let hiddenColumnsByBand;
+    return result;
+  }
 
-          columns = columns || that._columns;
+  getColumnId(column) {
+    if (column.command && column.type === GROUP_COMMAND_COLUMN_NAME) {
+      if (isCustomCommandColumn(this, column)) {
+        return `type:${column.type}`;
+      }
 
-          each(columns, (_, column) => {
-            if (column.ownerBand !== bandColumnIndex) {
-              return;
-            }
-            if (column.isBand) {
-              if (!column.visible) {
-                hiddenColumnsByBand = that.getChildrenByBandColumn(column.index);
-              } else {
-                hiddenColumnsByBand = that.getInvisibleColumns(that.getChildrenByBandColumn(column.index), column.index);
-              }
+      return `command:${column.command}`;
+    }
 
-              if (hiddenColumnsByBand.length) {
-                result.push(column);
-                result = result.concat(hiddenColumnsByBand);
-              }
-              return;
-            }
-            if (!column.visible) {
-              result.push(column);
-            }
-          });
+    return column.index;
+  }
 
-          return result;
-        },
-        getChooserColumns(getAllColumns) {
-          const columns = getAllColumns ? this.getColumns() : this.getInvisibleColumns();
-          const columnChooserColumns = columns.filter((column) => column.showInColumnChooser);
+  getCustomizeTextByDataType(dataType) {
+    return getCustomizeTextByDataType(dataType);
+  }
 
-          const sortOrder = this.option('columnChooser.sortOrder');
+  getHeaderContentAlignment(columnAlignment) {
+    const rtlEnabled = this.option('rtlEnabled');
 
-          return sortColumns(columnChooserColumns, sortOrder);
-        },
-        allowMoveColumn(fromVisibleIndex, toVisibleIndex, sourceLocation, targetLocation) {
-          const that = this;
-          const columnIndex = getColumnIndexByVisibleIndex(that, fromVisibleIndex, sourceLocation);
-          const sourceColumn = that._columns[columnIndex];
+    if (rtlEnabled) {
+      return columnAlignment === 'left' ? 'right' : 'left';
+    }
 
-          if (sourceColumn && (sourceColumn.allowReordering || sourceColumn.allowGrouping || sourceColumn.allowHiding)) {
-            if (sourceLocation === targetLocation) {
-              if (sourceLocation === COLUMN_CHOOSER_LOCATION) {
-                return false;
-              }
+    return columnAlignment;
+  }
+}
 
-              // @ts-expect-error
-              fromVisibleIndex = isObject(fromVisibleIndex) ? fromVisibleIndex.columnIndex : fromVisibleIndex;
-              // @ts-expect-error
-              toVisibleIndex = isObject(toVisibleIndex) ? toVisibleIndex.columnIndex : toVisibleIndex;
+export const columnsControllerModule: import('../module_types').Module = {
```

Comment:
```suggestion
export const columnsControllerModule: Module = {
```
it's better not to use `import()` statement